### PR TITLE
[MRG] added predict_proba for multinomial

### LINKED ISSUE #66: enforce same shape for input / output in multinomial case
This issue was discovered here: https://github.com/pavanramkumar/pyglmnet/commit/749bba9559bcddc65b0c5f68b7196691a5b5f489#commitcomment-17402965

basically if the output `y_pred` is going to be `n_samples x n_classes`, the input `y` should also be the same shape. Converting to this shape can be done by

1. the user 
2. using scikit-learn 
3. a convenience function for this can be provided.

but not inside `GLM` class.

### PR DESCRIPTION
**Pull request by pavanramkumar**
partially closes #66

### CODE REVIEW

**Comment by jasmainak** [comment]
@pavanramkumar do you think you could create PRs from another repo? It is better not to push to upstream. You can do:

```py
$ git remote add origin https://github.com/pavanramkumar/mypyglmnet/
$ git push origin predict_proba
```

That way, we can have a clean upstream repo ...

**Comment by pavanramkumar** [comment]
@jasmainak sure, i don't push directly to master; i make PRs from branches. 

what do you think about the diff? is the shape issue close to `scikit-learn` conventions? i'd like to tackle all the `version 0.1` issues asap.

**Comment by pavanramkumar** [comment]
merged this in the interest of moving forward.

**Comment by jasmainak** [comment]
@pavanramkumar I suppose I am a bit confused why

1. You have 2D output only for `predict_proba` for multinomial case but not for `predict`
2. `predict_proba` is implemented only for multinomial

**Comment by pavanramkumar** [comment]
@jasmainak this address the old issue that the erstwhile `predict()` function would return 2D output even though `fit()` accepts a `y` with 1D class labels.

With this fix, `predict()` returns a 1D output, while `predict_proba()` returns the 2D output corresponding to softmax probabilities for each class and each sample in the test set.

`predict_proba()` only makes sense for classification methods; in our case: `'multinomial'` and `'binomial'`. But for `'binomial'` `predict()` already returns a 1D output corresponding to class probability, so it's moot. I could potentially make `predict_proba()` return `yhat` and `1 - yhat`, but I  didn't see that to be a common use case.

**Comment by jasmainak** [comment]
okay, thanks for the explanation! yeah, it might be nice to have `predict_proba` work also for the `'binomial'` but we could also take a more pragmatic approach and wait for a user to complaint first :)
